loadflow/loadflow-validation/src/main/java/com/powsybl/loadflow/validation/ShuntCompensatorsValidation.java

```diff
@@ -0,0 +1,155 @@
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StaticVarCompensatorsValidation.class);
+
+    private ShuntCompensatorsValidation() {
+    }
+
+    public static boolean checkShunts(Network network, ValidationConfig config, Path file) throws IOException {
+        Objects.requireNonNull(file);
+        Objects.requireNonNull(config);
+        Objects.requireNonNull(file);
+        try (Writer writer = Files.newBufferedWriter(file, StandardCharsets.UTF_8)) {
+            return checkShunts(network, config, writer);
+        }
+    }
+
+    public static boolean checkShunts(Network network, ValidationConfig config, Writer writer) {
+        Objects.requireNonNull(network);
+        Objects.requireNonNull(config);
+        Objects.requireNonNull(writer);
+        try (ValidationWriter shuntsWriter = ValidationUtils.createValidationWriter(network.getId(), config, writer, ValidationType.SHUNTS)) {
+            return checkShunts(network, config, shuntsWriter);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    public static boolean checkShunts(Network network, ValidationConfig config, ValidationWriter shuntsWriter) {
+        Objects.requireNonNull(network);
+        Objects.requireNonNull(config);
+        Objects.requireNonNull(shuntsWriter);
+        LOGGER.info("Checking shunt compensators of network {}", network.getId());
+        return network.getShuntStream()
+                      .sorted(Comparator.comparing(ShuntCompensator::getId))
+                      .map(shunt -> checkShunts(shunt, config, shuntsWriter))
+                      .reduce(Boolean::logicalAnd)
+                      .orElse(true);
+    }
+
+    public static boolean checkShunts(ShuntCompensator shunt, ValidationConfig config, Writer writer) {
+        Objects.requireNonNull(shunt);
+        Objects.requireNonNull(config);
+        Objects.requireNonNull(writer);
+
+        try (ValidationWriter shuntsWriter = ValidationUtils.createValidationWriter(shunt.getId(), config, writer, ValidationType.SHUNTS)) {
+            return checkShunts(shunt, config, shuntsWriter);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    public static boolean checkShunts(ShuntCompensator shunt, ValidationConfig config, ValidationWriter shuntsWriter) {
+        Objects.requireNonNull(shunt);
+        Objects.requireNonNull(config);
+        Objects.requireNonNull(shuntsWriter);
+
+        boolean validated = true;
+        float p = shunt.getTerminal().getP();
+        float q = shunt.getTerminal().getQ();
+        int currentSectionCount = shunt.getCurrentSectionCount();
+        int maximumSectionCount = shunt.getMaximumSectionCount();
+        float bPerSection = shunt.getbPerSection();
+        float qMax = shunt.getProperties().containsKey("qMax") ? Float.parseFloat((String) shunt.getProperties().get("qMax")) : Float.NaN;
```

Comment:
It’s not a good idea to use this property. It exists only for a bijective import/export from and to the RTE study tool

Reply:
Is qMax available in any other way? Or should I skip the check on qMax (it is just for warning)?

Reply:
You should skip the test and remove the 3 relevant lines in the checkShunt method